memorymapping fails to build: src/fmemopen.c:62:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')

`memorymapping` fails to build as: https://hydra.nixos.org/build/153173858

```
src/fmemopen.c: In function 'seekfn':
src/fmemopen.c:62:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')
   62 |       if (offset >= 0) {
      |                  ^~
src/fmemopen.c:63:9: error: aggregate value used where an integer was expected
   63 |         pos = (size_t)offset;
      |         ^~~
src/fmemopen.c:70:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')
   70 |       if (offset >= 0 || (size_t)(-offset) <= mem->pos) {
      |                  ^~
```

### Notify maintainers

@veprbl 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: memorymapping
```

--- a/pkgs/development/libraries/memorymapping/default.nix
+++ b/pkgs/development/libraries/memorymapping/default.nix
@@ -37,7 +37,9 @@ stdenv.mkDerivation {
     description = "fmemopen for Mac OS and iOS";
     license = licenses.asl20;
     maintainers = with maintainers; [ veprbl ];
-    platforms = platforms.unix;
+    # Uses BSD-style funopen() to implement glibc-style fmemopen().
+    # Add more BSDs if you need to.
+    platforms = platforms.darwin;
     broken = stdenv.isAarch64;
   };
 }

